src/wavesurfer.js

```diff
@@ -1624,17 +1624,22 @@ export default class WaveSurfer extends util.Observer {
             peaks,
             val => Math.round(val * accuracy) / accuracy
         );
-        return new Promise((resolve, reject) => {
-            const json = JSON.stringify(arr);
-
-            if (!noWindow) {
-                window.open(
-                    'data:application/json;charset=utf-8,' +
-                        encodeURIComponent(json)
-                );
-            }
+        
+        new Promise((resolve, reject) => {
+            const json = arr;
             resolve(json);
-        });
+        }).then(response => {
+            if (!noWindow) {
+                window.open(URL.createObjectURL(
+                    new Blob([JSON.stringify(json)], {
+                        type: 'application/json;charset=utf-8'
+                    })
+                ));
+                return;
```

Comment:
@katspaugh I'm thinking of returning the actual blob here or maybe the same json of the peaks instead? What you think or should I add the blob export as an option?